packages/plugins/rudder_plugin_interface/lib/src/models/rudder_config.dart

```diff
@@ -152,51 +149,98 @@ class RudderConfig {
-      _webConfigMap["queueOptions"] = queueOpts;
-
       _webConfigMap["useBeacon"] = webConfig.useBeacon;
+      _webConfigMap["lockIntegrationsVersion"] =
+          webConfig.lockIntegrationsVersion;
+      _webConfigMap["lockPluginsVersion"] = webConfig.lockPluginsVersion;
+      _webConfigMap["polyfillIfRequired"] = webConfig.polyfillIfRequired;
+      _webConfigMap["useGlobalIntegrationsConfigInEvents"] =
+          webConfig.useGlobalIntegrationsConfigInEvents;
+      _webConfigMap["bufferDataPlaneEventsUntilReady"] =
+          webConfig.bufferDataPlaneEventsUntilReady;
+      _webConfigMap["dataPlaneEventsBufferTimeout"] =
+          webConfig.dataPlaneEventsBufferTimeout;
+      _webConfigMap["useServerSideCookies"] = webConfig.useServerSideCookies;
+      _webConfigMap["sendAdblockPage"] = webConfig.sendAdblockPage;
+      _webConfigMap["sameDomainCookiesOnly"] = webConfig.sameDomainCookiesOnly;
+
+      if (webConfig.queueOptions != null) {
+        _webConfigMap["queueOptions"] = webConfig.queueOptions?.toMap();
+      }
       //beacon queue opts if available
-      if (webConfig.useBeacon) {
-        beaconOpts["maxItems"] =
-            webConfig.maxBeaconItems.clamp(1, 1000).toString();
-        beaconOpts["flushQueueInterval"] =
-            webConfig.beaconFlushQueueInterval.clamp(1, 60000).toString();
-        _webConfigMap["beaconQueueOpts"] = beaconOpts;
+      if (webConfig.useBeacon && webConfig.beaconQueueOptions != null) {
+        _webConfigMap["beaconQueueOptions"] =
+            webConfig.beaconQueueOptions?.toMap();
       }
-      // cookie consent
-      if (webConfig.cookieConsentManagers != null) {
-        webConfig.cookieConsentManagers?.forEach((key, value) {
-          cookieConsent[key] = {"enabled": true};
-        });
-        _webConfigMap["cookieConsentManager"] = cookieConsent;
+      // consent manager
+      if (webConfig.consentManagement != null) {
+        _webConfigMap["consentManagement"] =
+            webConfig.consentManagement?.toMap();
       }
-      if (Utils.isValidUrl(webConfig.destSDKBaseURL)) {
-        _webConfigMap["destSDKBaseURL"] = webConfig.destSDKBaseURL;
-      } else {
-        RudderLogger.logWarn("Dest SDK Base Url is not valid, using default");
-        _webConfigMap["destSDKBaseUrl"] =
-            Constants.DEFAULT_DESTINATION_SDK_BASE_URL;
+      if (webConfig.sessions != null) {
+        _webConfigMap["sessions"] = webConfig.sessions?.toMap();
       }
-      sessionOpts['autoTrack'] = webConfig.autoSessionTracking;
-      if (webConfig.sessionTimeoutInMillis < 0) {
-        RudderLogger.logError("invalid sessionTimeoutInMillis. Set to default");
-        sessionOpts['timeout'] = Constants.DEFAULT_SESSION_TIMEOUT_WEB;
-      } else {
-        sessionOpts['timeout'] = webConfig.sessionTimeoutInMillis;
+      if (webConfig.destSDKBaseURL != null) {
+        if (Utils.isValidUrl(webConfig.destSDKBaseURL as String)) {
```

Comment:
URL validation is not required. It'll be repeated in the JS SDK.

Reply:
Done